### DISCUSSION TITLE
Initial network connectivity with WiFi?

### DISCUSSION THREAD

**Discussion by ishankhare07** [siderolabs/talos#8259] [Q&A]
I am using the metal iso to boot into the installer. The machine is a baremetal with a wifi adapter. Due to this I don't have the initial connectivity required to join the wifi network or connect by providing the connection configuration.

**Comment by smira** [comment]
Talos Linux doesn't support wireless networking.

**Reply by ryayon** [reply]
Hello @smira .

Can you explain why ?
I mean, is it a choice or a technical issue ?

**Reply by smira** [reply]
It's both:

* wireless is certainly not the best networking option for servers in general
* wireless configuration and management is a huge and complicated thing requiring lots of userspace processes to complete the task, and it blows up the scope of Talos

**Reply by ryayon** [reply]
Thank you for your answer that I, unfortunately, understand.

By the way, I could have also find this [answer](https://github.com/siderolabs/talos/discussions/6911#discussioncomment-12901717).

**Reply by centerionware** [reply]
Just enabling the kernel modules for wireless - (ideally) as an extension package, would make this possible as far as Talos goes I think. 

I've whipped up a container that does some magic and makes it probably possible to connect a wifi interface on a talos node using configuration that's node specific stored in the cluster as configmaps and secrets, it would just be run as a daemonset with rbac to read/create/list configmaps and secrets for the namespace the daemonset is run in.. basically just alpine, kubectl, wpa_supplicant, wireless-tools, and an entrypoint. Comes out to 25MB. I just need the kernel modules and a machine with wifi to test. likely might work on any kubernetes host with wireless drivers available. i tried to set it up to read /proc/cmdline too so it can extract an ssid and password value from the kernel command line but have yet to really be able to test anything beyond compiling the script and running it in an alpine test container running on a talos worker node to make sure the entrypoint gets all the way to the startup of wpa_supplicant.

While wifi isn't ideal, it can have real world use cases for worker nodes, such as display kiosks around a building/business. A good percentage don't get hardwired and current solutions tend to have them run windows in kiosk mode over wifi with vnc or some crazyness. I'd rather be able to offer talos and kubernetes to do the same thing and if customer wanted to update one or all of the kiosks displays they can do that in kubernetes somehow (Maybe a slick webui? Maybe rancher? idc how) instead of vnc to each machine. 

---
"Cattle Not Pets" - probably Bill Baker

**Reply by carlpaten** [reply]
I'm considering wifi as a fallback network in case of a switch outage - I'll try enabling the relevant kernel module. Thanks!
